lib/Profile.dart

```diff
@@ -421,182 +405,189 @@ class MapScreenState extends State<ProfilePage>
+    return ListView(
+      physics: NeverScrollableScrollPhysics(),
+      padding: const EdgeInsets.symmetric(vertical: 12, horizontal: 24),
+      shrinkWrap: true,
+      children: carPlates.map((plate) {
+        return Padding(
+          padding: const EdgeInsets.only(top: 12.0),
+          child: CarPlatePicker(
+            editable: false,
+            initialPlate: plate,
+            onChanged: (_) {},
+          ),
+        );
+      }).toList(),
     );
   }
 }
 
-class BusinessUnitWidget extends StatelessWidget {
-  final void Function(String value) businessUnitController;
-  final void Function(bool value) asyncCallController;
-  final String userBu;
-  const BusinessUnitWidget({
+class EditingCarPlatesList extends StatefulWidget {
+  final List<String> initialPlates;
+  final Function(List<String> plate) onChanged;
+  const EditingCarPlatesList({
     Key key,
-    @required this.businessUnitController,
-    @required this.asyncCallController,
-    this.userBu,
+    this.initialPlates,
+    @required this.onChanged,
   }) : super(key: key);
 
-  void onChange(String val) {
-    businessUnitController(val);
+  @override
+  _EditingCarPlatesListState createState() => _EditingCarPlatesListState();
+}
+
+class _EditingCarPlatesListState extends State<EditingCarPlatesList> {
+  List<String> carPlates;
+  @override
+  void initState() {
+    carPlates = [];
+    carPlates.addAll(widget.initialPlates); // deep copy
+    super.initState();
   }
 
-  void onLoaded(bool val) {
-    asyncCallController(val);
+  void changePlate(index, plate) {
+    if (index > carPlates.length - 1) {
+      carPlates.add(plate);
+    } else {
+      carPlates[index] = plate;
+    }
+    widget.onChanged(carPlates);
   }
 
   @override
   Widget build(BuildContext context) {
-    final width = MediaQuery.of(context).size.width;
-    final height = MediaQuery.of(context).size.height;
-    return Padding(
-        padding: EdgeInsets.only(
-            left: width * 0.08, right: width * 0.08, top: height * 0.01),
-        child: Column(
-          children: [
-            Container(
-              width: width * 0.85,
-              child: BusinessUnit(
-                passValue: onChange,
-                asyncCallController: onLoaded,
-                userBU: userBu,
+    return Column(
+      children: [
+        ListView(
+          physics: NeverScrollableScrollPhysics(),
+          padding: const EdgeInsets.only(top: 12, left: 24),
+          shrinkWrap: true,
+          children: carPlates.map((plate) {
+            int index = carPlates.indexOf(plate);
+            return Padding(
+              padding: const EdgeInsets.only(top: 12.0),
+              child: Row(
+                children: [
+                  Expanded(
+                    child: Container(
+                      child: CarPlatePicker(
+                        key: ObjectKey(
+                          plate, // prevents incorrect plate being deleted
+                        ),
```

Comment:
@welo10 After some research regarding our discussion on the use of keys in #83, I found that when removing an item from a list view by its index, not using a key will remove the item from the list itself, but will not update the widget in the `ListView`. 

Also, I used the `plate` as the key so that it is unique for every car plate. If the user entered the same car plates in both widgets then we don't care which widget we remove. 

I hope I could explain what I understood :sweat_smile: You can checkout this [video](https://www.youtube.com/watch?v=kn0EOS-ZiIc) that explains Keys in general as well as their benefit in our use case.